### PR TITLE
Rename RBE setup Github action and use RBE for windows CI runs

### PR DESCRIPTION
**Pull request by josephperrott**

### CODE REVIEW

File: github-actions/bazel/configure-remote/setup_remote_exec.sh
Comment (devversion) [inline github-actions/bazel/configure-remote/setup_remote_exec.sh:5]: Do you re-deploy a new orb version? we should do that:

1. Because sometimes this shell script is too long and doesn't work (especially on Windows shell limits)
2. If we don't do the update now, we end up with weird situations like in the past e.g. where we the service key had to be rotated for security reasons- and the updates ended up being extremely frustrating

Comment (devversion) [reply github-actions/bazel/configure-remote/setup_remote_exec.sh:5]: In general, we should not put NodeJS scripts in Shell in github actions because the whole brittle shell mechanism only existed for CircleCI

**Review by devversion — APPROVED** [review]
@josephperrott LGTM. Can we please at some point look into updating to these new action names in the primary repos? I don't think we get updates after the renames

File: github-actions/bazel/configure-remote/setup_remote_exec.sh
Comment (josephperrott) [reply github-actions/bazel/configure-remote/setup_remote_exec.sh]: I did not deploy a new orb version as I was waiting until it was merged.

I refactored things around to no longer use the a shell script for the github action.

**Comment by josephperrott** [comment]
@devversion  A bit of overhaul, PTAL

**Review by devversion — APPROVED** [review]
Nice! we should look into updating the action usages/orb to not incur future complications when we e.g. need to rotate a key./

**Comment by josephperrott** [comment]
This PR was merged into the repository by commit 6afa7d0c9e20968c74e1d3700c2fe76d31d8a72f.

**Pull request closed without merging**
